apps/angular/src/core/interceptors/refresh-token.interceptor.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import {
+	HttpEvent,
+	HttpInterceptor,
+	HttpHandler,
+	HttpRequest,
+	HttpErrorResponse,
+	HttpContextToken,
+} from '@angular/common/http';
+import { Observable, catchError, switchMap, tap, throwError } from 'rxjs';
+
+import { StorageService } from '../services/auth-storage.service';
+import { AuthService } from '../services/auth.service';
+
+/** Context for request to skip certain interceptors. */
+export const BYPASS_LOG = new HttpContextToken<boolean>(() => false);
+
+/** Interceptor to handle auth tokens. */
+@Injectable()
+export class RefreshTokenInterceptor implements HttpInterceptor {
+	public constructor(private readonly storage: StorageService, private readonly auth: AuthService) {}
+
+	/** @inheritdoc */
+	public intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+		return next.handle(request).pipe(
+			catchError((e: unknown) => {
+				// Check if a request has a bypass flag for this interceptor.
+				if (request.context.get(BYPASS_LOG) === true) {
+					return next.handle(request);
+				}
+
+				const refresh = this.storage.getRefreshToken();
+				if (e instanceof HttpErrorResponse) {
+					if (refresh && e.url !== 'https://api.camp-js.saritasa.rocks/api/v1/auth/token/refresh/') {
```

Comment:
Isn't part of that url in your env file?
```suggestion
					if (refresh && e.url !== `${this.env.ApiUrl}/auth/token/refresh/`) {
```